src/figuresClasses.ts

```diff
@@ -1,11 +1,85 @@
-export interface Figure {}
+export interface Figure {
+  shape: string;
+  color: string;
 
-export class Triangle implements Figure {}
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape = 'triangle';
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Fail data!!!');
+    }
+
+    const arr: number[] = [a, b, c];
+
+    arr.sort((x: number, y: number) => x - y);
+
+    if (arr[2] >= arr[0] + arr[1]) {
+      throw new Error('Fail data!!!');
+    }
+  }
+
+  getArea(): number {
+    const semiPerimeter = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(
+      semiPerimeter *
+        (semiPerimeter - this.a) *
+        (semiPerimeter - this.b) *
+        (semiPerimeter - this.c),
+    );
+
+    return Math.floor(area * 100) / 100;
+  }
+}
 
-export class Circle implements Figure {}
+export class Circle implements Figure {
+  public shape = 'circle';
 
-export class Rectangle implements Figure {}
+  constructor(
+    public color: string,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Fail data!!!');
+    }
+  }
+
+  getArea(): number {
+    const area: number =
+      Math.floor(Math.PI * this.radius * this.radius * 100) / 100;
+
+    return area;
+  }
+}
+
+export class Rectangle implements Figure {
+  public shape = 'rectangle';
+
+  constructor(
+    public color: string,
+    public width: number,
+    public height: number,
+  ) {
+    if (width <= 0 || height <= 0) {
```

Comment:
[CODE STYLE] - The error message should be more detailed. A suggested message is 'Rectangle width and height must be positive numbers.'